### DISCUSSION TITLE
Temperature & Humidity sensors; How do calibration work? (two different calibration setting available)

### DISCUSSION THREAD

**Discussion by marcelhoogantink** [Koenkk/zigbee2mqtt#18172] [Devices]
Hi,

I have some T&H sensors (Tuya  ZG-227Z, detected by Z2M as ZG-227ZL but working OK , and I need to calibrate the temperature a little.

Now I see different calibration options at two places:
1) Under 'Show':
![afbeelding](https://github.com/Koenkk/zigbee2mqtt/assets/51886888/f40042a6-e6c6-4db5-92ee-472a14dcca75)

2) Under 'Settings (specific)'
![afbeelding](https://github.com/Koenkk/zigbee2mqtt/assets/51886888/2d8f6ec7-6717-4666-a141-bd996de82e69)

Bot settings are separated, I can enter different values and they are saved.

I now wonder what these settings are for, are they both used, or only one. And if they are used both what is the difference?

**Comment by lux73** [comment]
must be a faulty implementation with your specific Devices

with my Xiaomi Temperature Sensors i have only unter **Settings (specific)** Tab Entrys for Calibration & Precision

so i would suggest to use this

**Comment by marcelhoogantink** [comment]
I understand, but when i try changing the values under 'Show' the values are changing also.
(An the is a bug: when I change the temperature calibration, the humidity calibration show the same value when I visit this page-tab again after leaving it.

**Comment by alexsotoaguilera** [comment]
When I change the value in the ‘Exposes’ tab, the reported 'State' data changes adding values in 'humidity_calibration' and 'temperature_calibration'. But reported humidity and temperature remains unchanged, the offset is not applied.

```
{
    "battery": 100,
    "humidity": 53,
    "humidity_calibration": 4.5,
    "last_seen": "2024-10-23T04:37:34+00:00",
    "linkquality": 184,
    "temperature": 23.1,
    "temperature_calibration": 0,
    "temperature_unit": "celsius"
}
```

When I change the value in the ‘Settings (specific)’ tab, the reported 'State' data don't changes. But the device in config file add the parameters.  Reported humidity and temperature applies the offset.

```
{
    "battery": 100,
    "humidity": 58,
    "humidity_calibration": 4.5,
    "last_seen": "2024-10-23T04:49:58+00:00",
    "linkquality": 255,
    "temperature": 34.9,
    "temperature_calibration": 0,
    "temperature_unit": "celsius"
}

  '0xa4c1385f2a74d3d1':
    friendly_name: Temperature and humidity sensor - Bathroom
    temperature_calibration: 10
    humidity_calibration: ''
```

How should it work? 

What are the changes in the ‘Exposes’ tab for?
